watertap/examples/flowsheets/lsrro/lsrro.py

```diff
@@ -453,77 +807,276 @@ def func_initialize(unit):
         pump.deltaP.setlb(0)
+        if idx > m.fs.Stages.first():
+            pump.max_lsrro_pressure_con = Constraint(
+                expr=(
+                    m.fs.lsrro_min_pressure,
+                    pump.control_volume.properties_out[0].pressure,
+                    m.fs.lsrro_max_pressure,
+                )
+            )
+            iscale.constraint_scaling_transform(
+                pump.max_lsrro_pressure_con,
+                iscale.get_scaling_factor(
+                    pump.control_volume.properties_out[0].pressure
+                ),
+            )
+        else:
+            pump.max_ro_pressure_con = Constraint(
+                expr=(
+                    m.fs.ro_min_pressure,
+                    pump.control_volume.properties_out[0].pressure,
+                    m.fs.ro_max_pressure,
+                )
+            )
+            iscale.constraint_scaling_transform(
+                pump.max_ro_pressure_con,
+                iscale.get_scaling_factor(
+                    pump.control_volume.properties_out[0].pressure
+                ),
+            )
+
+    if value(m.fs.NumberOfStages) > 1:
+        m.fs.EnergyRecoveryDevices[1].control_volume.properties_out[0].pressure.unfix()
+        m.fs.EnergyRecoveryDevices[1].deltaP.setub(0)
 
     # unfix eq pumps
-    for pump in m.fs.BoosterPumps.values():
+    for idx, pump in m.fs.BoosterPumps.items():
         pump.control_volume.properties_out[0].pressure.unfix()
-        pump.control_volume.properties_out[0].pressure.setlb(10e5)
-        pump.control_volume.properties_out[0].pressure.setub(85e5)
+        pump.max_ro_pressure_con = Constraint(
+            expr=(
+                m.fs.ro_min_pressure,
+                pump.control_volume.properties_out[0].pressure,
+                m.fs.ro_max_pressure,
+            )
+        )
+        iscale.constraint_scaling_transform(
+            pump.max_ro_pressure_con,
+            iscale.get_scaling_factor(pump.control_volume.properties_out[0].pressure),
+        )
         pump.deltaP.setlb(0)
 
+    if B_case == BCase.single_optimum:
+        m.fs.B_comp_system = Var(
+            domain=NonNegativeReals,
+            units=pyunits.m * pyunits.s**-1,
+            doc="Solute permeability coeff. constant in all LSR stages",
+        )
+        m.fs.B_comp_system.set_value(
+            m.fs.ROUnits[m.fs.LSRRO_Stages.first()].B_comp[0, "NaCl"]
+        )
+        m.fs.B_comp_system.setlb(3.5e-8)
+        m.fs.B_comp_system.setub(3.5e-8 * 1e2)
+    if A_case == ACase.single_optimum:
+        m.fs.A_comp_system = Var(
+            domain=NonNegativeReals,
+            units=pyunits.m * pyunits.s**-1 * pyunits.Pa**-1,
+            doc="Water permeability coeff. constant in all LSR stages",
+        )
+        m.fs.A_comp_system.set_value(m.fs.ROUnits[2].A_comp[0, "H2O"])
+        m.fs.A_comp_system.setlb(2.78e-12)
+        m.fs.A_comp_system.setub(4.2e-11)
```

Comment:
Not sure if we want to add parameters for A value bounds for parameter sweep. 

With regard to the A value itself, I think @eyoung55 added a parameter for the water permeability coefficient value on his local branch when running the param sweep analysis, which wasn't included on my branch.

Reply:
You need this parameter if you want to have a single value for `A_comp` across all LSRRO stages.

Regarding the bounds, it isn't clear to me that `A_comp_system` *or* `B_comp_system` need bounds at all -- they should already be implicitly bounded through the equality constraints tying them to the individual `A_comp` and `B_comp` variables on each RO unit.